src/mame/machine/p2000t.cpp

```diff
@@ -154,4 +154,32 @@ void p2000t_state::p2000t_port_707f_w(uint8_t data) { m_port_707f = data; }
 
 void p2000t_state::p2000t_port_888b_w(uint8_t data) {}
 void p2000t_state::p2000t_port_8c90_w(uint8_t data) {}
-void p2000t_state::p2000t_port_9494_w(uint8_t data) {}
+
+
+void p2000t_state::p2000t_port_9494_w(uint8_t data) {
+	//  The memory region E000-FFFF (8k) is bank switched
+	int available_banks = (m_ram->size() - 0xe000) / 0x2000;
+	if (data < available_banks)
+		membank("bank1")->set_entry(data);
+}
+
+void p2000t_state::init_p2000()
+{
+        m_program = &m_maincpu->space(AS_PROGRAM);
```

Comment:
Does this need to be stored in a member?  You’re only using it here on initialisation, you could just keep it in a local variable.

Reply:
will fix.